salt/states/dracr.py

```diff
@@ -0,0 +1,87 @@
+# -*- coding: utf-8 -*-
+'''
+.. versionadded:: Fluorine
+
+Management of Dell DRAC
+
+The DRAC module is used to create and manage DRAC cards on Dell servers
+
+Ensure the property is set
+
+  .. code-block:: yaml
+
+  test:
+    dracr.property_present:
+       - properties:
+           System.ServerOS.HostName: "Pretty-server"
+           System.ServerOS.OSName: "Ubuntu 16.04"
+       - admin_password: calvin
+       - admin_root: myuser
+       - host: 10.10.10.1
+
+'''
+
+from __future__ import absolute_import, print_function, unicode_literals
+
+import salt.exceptions
+
+
+def __virtual__():
+    '''
+    Ensure the racadm command is installed
+    '''
+    if salt.utils.which('racadm'):
+        return True
+
+    return False
+
+
+def property_present(properties, admin_username='root', admin_password='calvin', host=None, **kwargs):
+    '''
+    properties = {}
+    '''
+
+    ret = {'name': host,
+           'context': {'Host': host},
+           'result': True,
+           'changes': {},
+           'comment': ''}
+
+    if host is None:
+        host = __salt__['cmd.run_all']('ipmitool lan print | tr -d " " | grep "IPAddress:" | cut -f 2 -d ":"',
```

Comment:
I would rather pull all this data back, and do the processing in python.

Reply:
Hi @gtmanfred, is there special reason for this? I used shell mainly because I do not really see point in processing the output in python as it is plain text prefixed with plenty of spaces and no standard format. So we would basicly need to iterate over text and find regex with re module per line. 
Should I change it anyway?

Reply:
We are writing python, I would rather use the builtin python stuff, in the really weird case that tr, cut, or grep is not in the path for some reason, we know for certain that the python regex module will be there.

Reply:
Ok, it makes sense.

Reply:
@gtmanfred check now please.